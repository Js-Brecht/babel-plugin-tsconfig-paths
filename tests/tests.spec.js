const path = require('path')
const pluginTester = require('babel-plugin-tester').default
const tsconfigPaths = require('../lib')

const fixturePath = path.join.bind(path, __dirname, '__fixtures__')

pluginTester({
  plugin: tsconfigPaths,
  pluginName: 'babel-resolve-tsconfig-paths',
  title: 'TsConfig Paths Resolver',
  babelOptions: require(fixturePath('babel.config')),

  pluginOptions: {
    rootDir: fixturePath()
  },

  tests: {
    'Performs standard resolutions correctly': {
      pluginOptions: {
        rootDir: fixturePath('standard-resolutions'),
        tsconfig: fixturePath('standard-resolutions', 'tsconfig.json')
      },
      fixture: fixturePath('standard-resolutions', 'src', 'index.ts'),
      snapshot: true
    },

    'Resolves aliases of modules with directories of the same name': {
      pluginOptions: {
        rootDir: fixturePath('module-directory-same-name'),
        tsconfig: fixturePath('module-directory-same-name', 'tsconfig.json')
      },
      fixture: fixturePath('module-directory-same-name', 'src', 'index.ts'),
      snapshot: true
    },

    'Parses tsconfig with comments': {
      pluginOptions: {
        rootDir: fixturePath('tsconfig-comments'),
        tsconfig: fixturePath('tsconfig-comments', 'tsconfig.json')
      },
      fixture: fixturePath('tsconfig-comments', 'src', 'index.ts'),
      snapshot: true
    }
  }
})
