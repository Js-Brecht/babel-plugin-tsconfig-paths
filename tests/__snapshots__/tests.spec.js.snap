// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TsConfig Paths Resolver Parses tsconfig with comments: Parses tsconfig with comments 1`] = `

import test from '~/test'

console.log(test)

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _test = _interopRequireDefault(require("./test"));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

console.log(_test.default);


`;

exports[`TsConfig Paths Resolver Performs standard resolutions correctly: Performs standard resolutions correctly 1`] = `

import test from '~/test'

console.log(test)

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _test = _interopRequireDefault(require("./test"));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

console.log(_test.default);


`;

exports[`TsConfig Paths Resolver Resolves aliases of modules with directories of the same name: Resolves aliases of modules with directories of the same name 1`] = `

import test from '~/test'
import test2 from '~/test2'
import test3 from '~/test3'

console.log(test)
console.log(test2)
console.log(test3)

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _test = _interopRequireDefault(require("./test"));

var _index = _interopRequireDefault(require("./test2/index"));

var _test2 = _interopRequireDefault(require("~/test3"));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

console.log(_test.default);
console.log(_index.default);
console.log(_test2.default);


`;
